### ISSUE TITLE
The images generated by Cycle-GAN

### ISSUE THREAD

**Issue by huangyuyu0426** [Simon4Yan/Learning-via-Translation#5]
Hi,I use Cycle-GAN to generate images from Market to Duke.But the results are bad.I find the results in your paper are nice.Could you tell me some parameter settings e.g. epoch,learning rate,image size ...

**Comment by Simon4Yan** [comment]
Thanks for your attention. 

We mainly follow the original settings in CycleGAN paper. And we find that using instance norm is much better than batch norm. We provide instance norm code in our tensorflow project (see [here)](https://github.com/Simon4Yan/Learning-via-Translation/blob/64fa8f44eb9480174a7a51bb4b7628b1a327e185/SPGAN/Utils/ops.py#L84). Moreover, you can have a look at cyclegan project ([here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/64bba81a6867ad72aad461e27221ea278d25f5c1/models/networks.py#L15)).

Besides, I recommend you add the identity loss to preserve the color composition of input image.

**Comment by huangyuyu0426** [comment]
Thank you for your reply. I follow the original settings in CycleGAN paper and use instance norm in the project[](https://github.com/XHUJOY/CycleGAN-tensorflow). But when the epoch runs to 90, I find my generated images are not good as yours. Is it possible that my training data including 5000 Market1501 images and 5000 Duke are not enough? And how many images for training did you use?
Any reply will be appreciated.

**Comment by huangyuyu0426** [comment]
Thank you for your reply. I follow the original settings in CycleGAN paper and use instance norm in the [project.](https://github.com/XHUJOY/CycleGAN-tensorflow) But when the epoch runs to 90, I find my generated images are not good as yours. Is it possible that my training data including 5000 Market1501 images and 5000 Duke are not enough? And how many images for training did you use?
Any reply will be appreciated.

**Comment by Simon4Yan** [comment]
We use  all training images of duke and market dataset. We do not use gallery or probe images. Moreover, we train cycleGAN for 5 epochs (6epochs is okay) and we find that trying more epochs is not good.

**Comment by Simon4Yan** [comment]
@huangyuyu0426 I think you meet a similar case that the generated images seem to the inverted version of input images. If you meet this case, I think you can stop training and run again. Besides, I recommend you try identity loss to train cycleGAN, since identity loss will preserve the color composition of input images. 

Good luck to you!

**Comment by Simon4Yan** [comment]
@chl916185 thanks for your suggestion
